### DISCUSSION TITLE
docker installation

### DISCUSSION THREAD

**Discussion by t-redd** [imk-toolkit/imk-toolkit#50] [Show and tell]
Just a tip: I had problems to install docker on my linux box (ubuntu 20.04 LT). I tried all variants I could find (from docker.com direct and via deb repos, from ubuntu and via snap), all gave an error when starting the dockerd. Finally, I used the docker repo again, which failed again, did a reboot, and then it worked.

**Comment by ucyo** [comment]
Maybe there was a problem with the group permissions. Usually there are there [post installation step](https://docs.docker.com/engine/install/linux-postinstall/) which help with running the containers. But if this was the only problem during installation I am actually pretty happy 😄

**Reply by FObersteiner** [reply]
it wasn't a joy on Windows either; had to dig deep into the "Computerverwaltung" to set required permissions ;-)

**Reply by ucyo** [reply]
Interesting. Has this occurred with other docker libraries? Or is this specific for our library?

**Reply by FObersteiner** [reply]
I meant the setup of docker itself. I did that a while ago to try something else, actually did't try with imktk yet 🙄

**Reply by ucyo** [reply]
Okay. That is at least somewhat comforting. Don't even try to install anaconda 🐍  It is a hassle 💢

**Reply by FObersteiner** [reply]
@ucyo a little update; building the container works on Windows, however the imktk methods don't seem to get recognized when running the example from the readme:
```python
>>> anomaly_free_t = t.imktk.anomalies()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/python/.local/lib/python3.9/site-packages/xarray/core/common.py", line 246, in __getattr__
    raise AttributeError(
AttributeError: 'DataArray' object has no attribute 'imktk'
```
It's all working fine on my Ubuntu machine...

Anyways, at least for me this is not actually an issue. `imktk` works fine if I do the old-school pip install.

**Comment by t-redd** [comment]
Just add that at least for my docker installation I have to start docker with sudo.

**Comment by ucyo** [comment]
There are recommended `post-installation` steps on the Docker webpage: https://docs.docker.com/engine/install/linux-postinstall/

**Comment by BorjaEst** [comment]
Is there a docker to run docker?

**Reply by ucyo** [reply]
It is possible to run docker in docker. You need to bind the outside process and it will work. But I do not think that this was the issue here.

**Reply by BorjaEst** [reply]
Sorry, it was just a joke I could not resist
